zorkatest.js

```diff
@@ -0,0 +1,417 @@
+
+async function play() {
+    //if plaer is poisoned, begin the countdown to failure
+  if (poison === true && poisonTimer > 0) {
+    --poisonTimer;
+    console.log(`time is ticking, use that antidote soon!`);
+  } else if (poison === true && poisonTimer <= 0) {
+    console.log(
+      "As the poison courses through your vision begins to fade and the room grows dark. You have failed in your quest"
+    );
+    process.exit();
+  }
+  //ask for user input
+  let userAction = await ask("what would you like to do?");
+  //sanitize and split user input
+  let inputArray = userAction.toLowerCase().split(" ");
+  //the first string of the user input becomes the action and the second becomes the target
+  let action = inputArray[0];
+  let target = inputArray[1];
+
+  //if the action is valid do something, with the target as the argument
+  if (action === "take") {
+    //set up exception for the key
+    if (itemTable[target].name === `key` && chestSearch === true) {
+      itemTable[target].take();
+      //player becomes poisoned after taking key
+      poison = true;
+      console.log(`
+        as you you pick up the key you feel a prick on your finger! It appears that the chest was trapped and you have been poisoned! Find an antidote quiskly or you will soon perish.
+        `);
+        //check if the target is valid so it doesn't error out trying to use an undefined argument
+    } else if(itemTable[target].name === 'treasure' && curLoc === 'vault') {
+        console.log(`You have succeeded in your quest. Revel in the fact that your days wallowing about in dark caves are over!`)
+        process.exit()
+    } else if (itemTable[target] instanceof Item) {
+      console.log(itemTable[target].take());
+    } else {
+      //if the action isn't valid throw an error
+      console.log("you cant pick that up!");
+    }
+  } else if (action === "help") {
+      //allow user to check available commands
+    console.log(`
+      move : move to target location. Eligble targets will be highlighted in [] brackets
+      take: take the targeted object, if it can be taken
+      use: use an object that you have in your inventory
+      drop: drop an object that you have in your inventory
+      examine: take a closer look at the targeted object, use this to interact with objects that you cannot take
+      lookaround: take a closer look at your surroundings, and list objects that can be interacted with
+      inventory: check your current inventory
+      
+      `);
+  } else if (action === "move") {
+    //prevent the player from going through the tunnel without first picking up a torch
+    if (
+      locationTable[target] instanceof Room &&
+      locationTable[target].location === `tunnel` &&
+      !playerInventory.includes("torch")
+    ) {
+      console.log(
+        `no way you can make it through that tunnel without a light source!`
+      );
+      //prevent player from just moving to the vualt
+    } else if (
```

Comment:
your comments in this file are great. I would like to see more of them above your conditional logic. Where they are, it helps me understand greatly. Where they aren't, I get the jist but would appreciate the care you clearly put in explaining things elsewhere! This is an example of a place they would be helpful.